staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -76,38 +77,85 @@ const (
 	zonesKey = "zones"
 )
 
-// replaceTopology overwrites an existing topology key by a new one.
+// replaceTopology overwrites an existing key by a new one.
+// If there are any newKey already exist in a term, we will combine the
+// replaced key Values with the existing ones
 func replaceTopology(pv *v1.PersistentVolume, oldKey, newKey string) error {
+	// Make sure the necessary fields exist
+	if pv == nil || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil ||
+		pv.Spec.NodeAffinity.Required.NodeSelectorTerms == nil || len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 0 {
+		return nil
+	}
 	for i := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
+		newKeyValsMap := make(map[string]bool)
+		var requirementsToRemove []int
 		for j, r := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions {
 			if r.Key == oldKey {
 				pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions[j].Key = newKey
 			}
+			if r.Key == newKey || r.Key == oldKey {
+				for _, v := range r.Values {
+					newKeyValsMap[v] = true
+				}
+				requirementsToRemove = append(requirementsToRemove, j)
+			}
+		}
+
+		// Combine any duplicated keys
+		if len(requirementsToRemove) > 1 {
+			// This means there are duplicated new Labels, we shall combine them together
```

Comment:
If only a single entry of the new key exists in NodeAffinity, based on the logic above, it would be appended to `requirementsToRemove` even though it's not a duplicate right?

Reply:
It will be added but the `if len(requirementsToRemove) > 1` logic here makes sure that we do not do unnecessary removal.